common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java

```diff
@@ -222,26 +308,155 @@ private void handleStateChange(LootrunningState oldState, LootrunningState newSt
+                if (newTaskPrediction.predictionScore() < oldPrediction.predictionScore()) {
+                    // The prediction is the same, but the score is better, so update.
+                    beacons.put(beacon.color(), newTaskPrediction);
+                }
+                return;
+            }
+
+            // The prediction is a location that an other color beacon is already at
+            Optional<TaskPrediction> usedTaskPredictionOpt = usedTaskLocations.stream()
+                    .filter(pair -> Objects.equals(pair.taskLocation(), closestTaskLocation))
+                    .findFirst();
+            if (usedTaskPredictionOpt.isPresent()) {
+                TaskPrediction usedTaskPrediction = usedTaskPredictionOpt.get();
+
+                // We predict that we are closer to the task location than the other beacon.
+                // Overwrite the other beacon's prediction.
+                if (newTaskPrediction.predictionScore() < usedTaskPrediction.predictionScore()) {
+                    beacons.put(beacon.color(), newTaskPrediction);
+                    McUtils.sendMessageToClient(Component.literal("Predicted " + beacon.color() + " beacon at "
+                            + beacon.position() + " to be closer to " + closestTaskLocation + " than "
+                            + usedTaskPrediction.beacon() + " (" + usedTaskPrediction.predictionScore() + " > "
+                            + newTaskPrediction.predictionScore() + ")"));
+
+                    // Update the other beacon's prediction.
+                    updateTaskLocationPrediction(usedTaskPrediction.beacon());
+                    break;
+                } else {
+                    // We predict that the other beacon is closer to the task location than us.
+                    // Use the second best prediction.
+                    continue;
+                }
+            }
+
+            // The prediction is not used by another beacon.
+            beacons.put(beacon.color(), newTaskPrediction);
+            break;
+        }
+
+        // Finally, update the markers.
+        LOOTRUN_BEACON_COMPASS_PROVIDER.reloadTaskMarkers();
+    }
+
+    private Pair<Double, TaskLocation> calculatePredictionScore(Beacon beacon, TaskLocation currentTaskLocation) {
+        // Player Location
+        Vector2d playerPosition = new Vector2d(
+                McUtils.player().position().x(), McUtils.player().position().z());
+        // Task Location
+        Vector2d taskLocationPosition = new Vector2d(
+                currentTaskLocation.location().x(),
+                currentTaskLocation.location().z());
+        // Wynn Beacon
+        Vector2d beaconPosition =
+                new Vector2d(beacon.position().x(), beacon.position().z());
+
+        // Short circuit if the beacon matches a task location.
+        // Wynn beacons are always at the center of a block, if they are in their "final" position.
```

Comment:
Can't it be that they happen to land on x.5, z.5 by pure coincidence, even if they are not final? Maybe that is so unlikely that we can ignore it...

Reply:
Really unlikely.. and it can't happen in the new PR.